include/point_cloud_transport/publisher_plugin.hpp

```diff
@@ -51,31 +52,39 @@ namespace point_cloud_transport
 class POINT_CLOUD_TRANSPORT_PUBLIC PublisherPlugin
 {
 public:
+  //! \brief Result of cloud encoding. Either an holding the compressed cloud, empty value or error message.
+  typedef cras::expected<std::optional<const std::shared_ptr<rclcpp::SerializedMessage>>, std::string> EncodeResult;
+
   PublisherPlugin() = default;
   PublisherPlugin(const PublisherPlugin &) = delete;
   PublisherPlugin & operator=(const PublisherPlugin &) = delete;
 
   //! Get a string identifier for the transport provided by this plugin
   virtual std::string getTransportName() const = 0;
 
-  /**
-   * \brief Advertise a topic, simple version.
-   */
+  //! \brief Advertise a topic, simple version.
   void advertise(
     std::shared_ptr<rclcpp::Node> nh,
     const std::string & base_topic,
     rmw_qos_profile_t custom_qos = rmw_qos_profile_default,
-    const rclcpp::PublisherOptions & options = rclcpp::PublisherOptions())
-  {
-    advertiseImpl(nh, base_topic, custom_qos, options);
-  }
+    const rclcpp::PublisherOptions & options = rclcpp::PublisherOptions());
 
   //! Returns the number of subscribers that are currently connected to this PublisherPlugin
   virtual uint32_t getNumSubscribers() const = 0;
 
   //! Returns the topic that this PublisherPlugin will publish on.
   virtual std::string getTopic() const = 0;
 
+  //! Return the datatype of the transported messages (as text in the form `package/Message`).
+  virtual std::string getDataType() const = 0;
+
+  /**
+   * \brief Encode the given raw pointcloud into EncodeResult
+   * \param[in] raw The input raw pointcloud.
```

Comment:
there is a mix of documentation styles. In same places we used `@param` and in other `\param`. 

Can we use `\param` ? which it's the one used in ROS 2?